epp-client/src/extensions/namestore.rs

```diff
@@ -0,0 +1,88 @@
+//! Types for EPP namestore request and responses
+
+use epp_client_macros::ElementName;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    common::{ElementName, StringValue},
+    request::EppExtension,
+};
+
+pub const XMLNS: &str = "http://www.verisign-grs.com/epp/namestoreExt-1.1";
+
+/// Type that represents the &lt;epp&gt; request for domain &lt;check&gt; command
+///
+/// ## Usage
+///
+/// ```no_run
+/// use std::collections::HashMap;
+///
+/// use epp_client::config::{EppClientConfig, EppClientConnection};
+/// use epp_client::EppClient;
+/// use epp_client::domain::check::DomainCheck;
+/// use epp_client::generate_client_tr_id;
+/// use epp_client::extensions::namestore::Namestore;
+///
+/// #[tokio::main]
+/// async fn main() {
+///     // Create a config
+///     let mut registry: HashMap<String, EppClientConnection> = HashMap::new();
+///     registry.insert(
+///         "registry_name".to_owned(),
+///         EppClientConnection {
+///             host: "example.com".to_owned(),
+///             port: 700,
+///             username: "username".to_owned(),
+///             password: "password".to_owned(),
+///             ext_uris: None,
+///             tls_files: None,
+///         },
+///     );
+///     let config = EppClientConfig { registry };
+///
+///     // Create an instance of EppClient, passing the config and the registry you want to connect to
+///     let mut client = match EppClient::new(&config, "registry_name").await {
+///         Ok(client) => client,
+///         Err(e) => panic!("Failed to create EppClient: {}",  e)
+///     };
+///
+///     let namestore_ext = Namestore::new("com");
+///
+///     // Create an DomainCheck instance
+///     let domain_check = DomainCheck::<Namestore>::new(
+///         vec!["eppdev-100.com", "eppdev-200.com"],
+///     ).with_extension(namestore_ext);
+///
+///     // send it to the registry and receive a response of type EppDomainCheckResponse
+///     let response = client.transact(domain_check, generate_client_tr_id(&client).as_str()).await.unwrap();
+///
+///     println!("{:?}", response);
+///
+///     client.logout().await.unwrap();
+/// }
+/// ```
+impl Namestore {
+    /// Create a new RGP restore report request
+    pub fn new(subproduct: &str) -> Namestore {
+        Namestore {
+            xmlns: XMLNS.to_string(),
+            subproduct: subproduct.into(),
+        }
+    }
+}
+
+impl EppExtension for Namestore {
+    type Response = Namestore;
+}
+
+#[derive(Serialize, Deserialize, Debug, ElementName)]
+#[element_name(name = "namestoreExt:namestoreExt")]
+/// Type for EPP XML &lt;namestoreExt&gt; extension
+pub struct Namestore {
```

Comment:
I realize this would somewhat conflict with the XML name, but maybe name this `NameStore`?

Reply:
Yeah, could go either way with this.

The RFC uses "Namestore": https://www.verisign.com/assets/epp-sdk/verisign_epp-extension_namestoreext_v01.html

However it is referenced as "NameStore" here: https://www.verisign.com/en_US/channel-resources/domain-registry-products/epp-sdks/index.xhtml

I think I would match what is used in the RFC

Reply:
I think the Rust naming guidelines should override the capitalization in the spec, otherwise it's very hard to remember the names correctly (quick, which type name has a different capitalization in the spec -- or even, which type name is described in the spec vs is something we invented here?). It seems like Verisign is also doing the same in their SDK.

Reply:
Fair enough! Changed to NameStore